src/test/java/org/assertj/core/data/Percentage_withPercentage_Test.java

```diff
@@ -14,27 +14,17 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
-import static org.assertj.core.api.Assertions.assertThatNullPointerException;
 import static org.assertj.core.data.Percentage.withPercentage;
 import static org.assertj.core.internal.ErrorMessages.percentageValueIsInRange;
 
 import org.junit.jupiter.api.Test;
 
 /**
- * Tests for {@link Percentage#withPercentage(Number)}.
+ * Tests for {@link Percentage#withPercentage(double)}.
  *
  * @author Alexander Bischof
  */
-public class Percentage_withPercentage_with_Double_Test {
-
-  @SuppressWarnings("null")
-  @Test
-  public void should_throw_error_if_value_is_null() {
```

Comment:
This test had actually still been working, but for the wrong reasons. I think the runtime tries to get a Double from the null, which it can then unbox to the double that the method actual requires.

So the runtime was throwing the NPE, instead of the `requireNonNull` check in the code (which is now deleted).